sound/soc/sof/intel/hda-dsp.c

```diff
@@ -508,9 +524,25 @@ int hda_dsp_runtime_suspend(struct snd_sof_dev *sdev)
 
 int hda_dsp_suspend(struct snd_sof_dev *sdev)
 {
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct pci_dev *pci = to_pci_dev(sdev->dev);
 	int ret;
 
+	if (sdev->s0_suspend) {
+		/* enable L1SEN to make sure the system can enter S0Ix */
+		hda->l1_support_changed =
+			snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR,
+						HDA_VS_INTEL_EM2,
+						HDA_VS_INTEL_EM2_L1SEN,
+						HDA_VS_INTEL_EM2_L1SEN);
```

Comment:
I don't understand the logic. Is this to preserve the state if the CONFIG_ENABLE_L1EN is set?
Please add comments, this is not self-explanatory why a new state variable is needed

Reply:
during suspend, we force to set the L1SEN bit, 
as already in comment message, the flag 'l1_support_changed' is used to indicate that "during suspend, is L1SEN changed or not", if it is changed(from 0 -> 1), we should restore it to be 0 during resuming, if not (from 1 ->1), we don't need to do anything at resuming.

We could use solution like what I do here to add this flag, or use another solution which iterate the list again to decide if we need clear the bit or not, something like this:
```
	list_for_each_entry(s, &bus->stream_list, list) {
		if (s->opened && s->direction == SNDRV_PCM_STREAM_CAPTURE)) {
			snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR,
						HDA_VS_INTEL_EM2,
						HDA_VS_INTEL_EM2_L1SEN,
						0);
                        break;
                }
	}
```
I previously considered the latter solution, and finally choose the former one, adding a bool memory space consuming is better to adding a list searching time consuming in Algorithm theory, agree?